pkg/operator/ceph/pool/controller.go

```diff
@@ -33,9 +34,11 @@ import (
 )
 
 const (
-	replicatedType         = "replicated"
-	erasureCodeType        = "erasure-coded"
-	poolApplicationNameRBD = "rbd"
+	replicatedType          = "replicated"
+	erasureCodeType         = "erasure-coded"
+	poolApplicationNameRBD  = "rbd"
+	poolDeleteRetryInterval = 10 // seconds
+	poolDeleteMaxRetries    = 15
```

Comment:
What if we didn't retry at all? If there are any images, we just don't delete the pool. Then when we have the override setting, we can force the delete if needed according to that setting.

Reply:
I can do that. and also one question related to cluster delete, even in cluster deletion we are doing retry, I though we can use the same logic here.

Reply:
Ok, the cluster deletion question really needs to be thought through at another level. The approach currently is that before we allow the cluster to be deleted, all the PVCs have to be removed that were created by the rook flex driver. This helps ensure ceph clients aren't hung if the cluster gets removed from beneath them. 

Now the question is really how we can approach this with the CSI driver. If we're just adding this check to the pool deletion, it's not sufficient. This is a good change and we should finish this PR, but we really need a different approach for the CSI cleanup question. We need to also implement this in the cluster deletion so we can prevent the cluster from being deleted if there are any PVs bound in the cluster. The question is how we can detect that they exist. Likely we can look at the PV properties to see the provisioner is ceph-csi.

Reply:
Continuing this, the [PR](https://github.com/rook/rook/pull/3967) for cluster deletion, takes the route of checking PVs that are in use by the cluster. This PR, for pools, takes the route of ensuring there are no images and snapshots on the pool.

IMO both are divergent methods, and we possibly need to use one of the 2 in both (and CephFS deletion) places.

@travisn , what is the requirement here? Is it to ensure the pool is empty, or pool is not in use by the kubernetes cluster and it does not matter if there are other images in the pool? If the former, the patch is along required lines, if latter, then the mentioned PR is along the required lines. If both requirements are different then the divergence is fair, and we can proceed.

Further, when deleting a cluster, does it invoke deleting each pool in the cluster and then deleting the cluster, or is it directly just deleting the cluster? (latter may be more efficient as we are losing the cluster anyway, and hence deleting individual pools and filesystems may not be required)